tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/api/traversal/Filter.java

```diff
@@ -0,0 +1,285 @@
+     * assets, and the root path.
+     *
+     * @param folderIncludes The patterns to include folders.
+     * @param assetIncludes  The patterns to include assets.
+     * @param folderExcludes The patterns to exclude folders.
+     * @param assetExcludes  The patterns to exclude assets.
+     * @param rootPath       The root path to apply the filter. This is used to remove the root path
+     *                       prefix from the folder and asset paths before applying the filter.
+     */
+    private Filter(
+            Set<PathMatcher> folderIncludes,
+            Set<PathMatcher> assetIncludes,
+            Set<PathMatcher> folderExcludes,
+            Set<PathMatcher> assetExcludes,
+            String rootPath) {
+        this.folderIncludes = folderIncludes;
+        this.assetIncludes = assetIncludes;
+        this.folderExcludes = folderExcludes;
+        this.assetExcludes = assetExcludes;
+        this.rootPath = rootPath;
+    }
+
+    /**
+     * Applies the filter to the given folder, filtering its sub-folders and assets based on the
+     * configured includes and excludes patterns.
+     *
+     * @param folder The folder to apply the filter to.
+     * @return The filtered folder view.
+     */
+    public FolderView apply(FolderView folder) {
+
+        List<FolderView> filteredSubFolders = new ArrayList<>();
+        List<AssetView> filteredAssets = new ArrayList<>();
+
+        if (folder.subFolders() != null) {
+            folder.subFolders().iterator().forEachRemaining(subFolder -> {
+
+                subFolder = validateFolder(subFolder);
+                filteredSubFolders.add(subFolder);
+            });
+        }
+
+        if (folder.assets() != null) {
+            folder.assets().versions().iterator().forEachRemaining(assetVersion -> {
+
+                var assetVersionPath = assetVersion.metadata().get(PATH_META_KEY.key()).toString();
+                var assetPath = Paths.get(assetVersionPath).
+                        resolve(assetVersion.name()).toString();
+
+                if (!rootPath.equals("/")) {
+                    assetPath = assetPath.replaceFirst("^" + rootPath, "");
+                }
+
+                if (useAsset(assetPath)) {
+                    filteredAssets.add(assetVersion);
+                }
+            });
+        }
+
+        folder = folder.withSubFolders(filteredSubFolders);
+        var versions = AssetVersionsView.builder().versions(filteredAssets).build();
+        return folder.withAssets(versions);
+    }
+
+    /**
+     * Validates the given folder by determining whether it should be included based on the includes
+     * and excludes patterns for folders setting specific flags in the {@link FolderView} object.
+     *
+     * @param subFolder The folder to validate.
+     * @return The validated folder view.
+     */
+    private FolderView validateFolder(FolderView subFolder) {
```

Comment:
not sure if validate is the right name, since it is not actually validating, since do not return any error.
it is mostly such as purge or clean

Reply:
Changed to `evaluateFolder`